pkg/webhook/admission/inferenceservice/inferenceservice_validator.go

```diff
@@ -43,9 +48,36 @@ func (validator *Validator) Handle(ctx context.Context, req admissiontypes.Reque
 		return admission.ErrorResponse(http.StatusInternalServerError, err)
 	}
 
+	if constants.IsEnabledMatchWorkingNamespaces {
+		if err := validator.validateNamespace(isvc, req.AdmissionRequest.Namespace); err != nil {
+			return admission.ErrorResponse(http.StatusBadRequest, err)
+		}
+	}
+
 	if err := isvc.ValidateCreate(validator.Client); err != nil {
 		return admission.ErrorResponse(http.StatusBadRequest, err)
 	}
 
 	return admission.ValidationResponse(true, "allowed")
 }
+
+func (validator *Validator) validateNamespace(isvc *kfserving.InferenceService, namespace string) error {
```

Comment:
I think this logic should not be in kfserving, here it is basically implementing a rbac rule. The right place should be profile controller where it can add a namespace scoped rolebinding to permit user to create inferenceservice if namespace is enabled for kfserving.

Reply:
I think we should keep a close loop for this feature in kfserving side no matter if kfserving is deployed in kubeflow or individually.
even we only deploy kfserving individually, we still can handle this case.

Reply:
Does it violate the principle :  "Protect this namespace selector under a environment variable which only enables in kubeflow installation, standalone kfserving installation shouldn't require this." - As long as it does not, we are good

Reply:
it doesn't violate the principle since it only check if the namespace is valid or not only when the corresponding envVar is set and value is `enabled` (standalone kfserving installation never set the env var).